sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -0,0 +1,403 @@
+    ///
+    /// The server provides healthcheck and version endpoints.
+    #[clap(
+        short,
+        long,
+        env = "ESPRESSO_COMMITMENT_TASK_PORT",
+        default_value = "8772"
+    )]
+    commitment_task_port: u16,
+
+    #[clap(flatten)]
+    sql: persistence::sql::Options,
+}
+
+#[async_std::main]
+async fn main() -> anyhow::Result<()> {
+    setup_logging();
+    setup_backtrace();
+
+    let opt = Args::parse();
+    let options = options::Options::from(options::Http {
+        port: opt.sequencer_api_port,
+    })
+    .status(Default::default())
+    .state(Default::default())
+    .submit(Default::default())
+    .query_sql(Default::default(), opt.sql);
+
+    let (url, _anvil) = if let Some(url) = opt.rpc_url {
+        (url, None)
+    } else {
+        tracing::warn!("L1 url is not provided. running an anvil node");
+        let instance = AnvilOptions::default().spawn().await;
+        let url = instance.url();
+        tracing::info!("l1 url: {}", url);
+        (url, Some(instance))
+    };
+
+    let network = TestNetwork::new(
+        options,
+        [persistence::no_storage::Options; TestConfig::NUM_NODES],
+        url.clone(),
+    )
+    .await;
+
+    let contracts = Contracts::new();
+
+    tracing::info!("deploying the contracts");
+    let contracts = deploy(
+        url.clone(),
+        opt.mnemonic.clone(),
+        opt.account_index,
+        true,
+        network.light_client_genesis(),
+        contracts,
+    )
+    .await?;
+
+    let hotshot_address = contracts
+        .get_contract_address(Contract::HotShot)
+        .expect("Cannot get the hotshot contract address");
+    tracing::info!("hotshot address: {}", hotshot_address);
+
+    tracing::info!("starting the commitment server");
+    start_commitment_server(opt.commitment_task_port, hotshot_address, SEQUENCER_VERSION).unwrap();
+
+    tracing::info!("starting the builder server");
+    let builder_address = "0xb0cfa4e5893107e2995974ef032957752bb526e9"
+        .parse()
+        .unwrap();
+    start_builder_server(opt.builder_port, builder_address, SEQUENCER_VERSION).unwrap();
+
+    let sequencer_url =
+        Url::parse(format!("http://localhost:{}", opt.sequencer_api_port).as_str()).unwrap();
+    let commitment_task_options = CommitmentTaskOptions {
+        l1_provider: url,
+        l1_chain_id: None,
+        hotshot_address,
+        sequencer_mnemonic: opt.mnemonic,
+        sequencer_account_index: opt.account_index,
+        query_service_url: Some(sequencer_url),
+        request_timeout: Duration::from_secs(5),
+        delay: None,
+    };
+
+    tracing::info!("starting hotshot commitment task");
+    run_hotshot_commitment_task::<es_version::SequencerVersion>(&commitment_task_options).await;
+
+    Ok(())
+}
+
+// In the test node binary, for now we don't need to run the builder. We just hardcode the builder address
+// and expose the needed endpoint.
+fn start_builder_server<Ver: StaticVersionType + 'static>(
```

Comment:
TestNetwork already runs the simple builder so we don't need to run it separately.